catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/AbstractSolrClientCommand.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.solr.offlinegazetteer;
+
+import static ddf.catalog.solr.offlinegazetteer.GazetteerConstants.COLLECTION_NAME;
+
+import java.io.IOException;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.codice.ddf.commands.catalog.SubjectCommands;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.client.solrj.UnavailableSolrException;
+import org.codice.solr.factory.SolrClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractSolrClientCommand extends SubjectCommands {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractSolrClientCommand.class);
+
+  @Reference protected SolrClientFactory clientFactory;
+
+  @Option(
+      name = "--force",
+      aliases = "-f",
+      description = "Force the command without a confirmation message.")
+  protected boolean force = false;
+
+  @SuppressWarnings({"java:S2139" /* Logging and rethrowing failure exception intentionally */})
+  @Override
+  protected Object executeWithSubject() throws Exception {
```

Comment:
I extracted the `force` option and the logic to check of the `SolrClient` is available to this class. I also made this extend `SubjectCommands` so that the `runWithSubjectOrElevate` logic didn't need to be included in these classes.